packages/create-redwood-app/src/create-redwood-app.js

```diff
@@ -33,7 +20,31 @@ import { name, version } from '../package'
  * - packages/create-redwood-app/src/create-redwood-app.js (this file)
  *
  */
+import chalk from 'chalk'
+import checkNodeVersion from 'check-node-version'
+import { prompt } from 'enquirer'
+import execa from 'execa'
+import fs from 'fs-extra'
+import { Listr, figures } from 'listr2'
+import terminalLink from 'terminal-link'
+import { hideBin } from 'yargs/helpers'
+import yargs from 'yargs/yargs'
+
+import { name, version } from '../package'
+
+import { startTelemetry, shutdownTelemetry } from './telemetry'
 ;(async () => {
+  //
+
+  // Telemetry
+  if (
+    !process.argv.includes('--no-telemetry') && // Must include '--no-telemetry' exactly because we want to do this check before any yargs. TODO: Communicate this on cmd help
```

Comment:
I had the same problem in the CLI's index.js file. I need the value of the `cwd` option before I invoke yargs. You should be able to use the same solution I did:

https://github.com/redwoodjs/redwood/blob/8569f5ff0d5996c13aa0b23194aed39a54acf9c0/packages/cli/src/index.js#L60

So in your case...

```ts
import { hideBin, Parser } from 'yargs/helpers'

// ...

const { telemetry } = Parser(hideBin(process.argv))
```

Then check if `telemetry` is `false` later, no need to deal with the "no-" prefix, etc.

Reply:
Excellent I had a suspicion that it might have been a similar issue you'd dealt. I'll try out your solution for sure.